js/packages/store/provider.tsx

```diff
@@ -1,363 +1,82 @@
-							name: payloadName,
-							payload: eventPayload,
-						})
-					}
-				})
-				cancel = await stream.start()
-			})
-			.catch((err) => {
-				if (err === EOF) {
-					console.info('end of the events stream')
-				} else {
-					console.warn('events stream error:', err)
-				}
-				dispatch({ type: 'SET_STREAM_ERROR', payload: { error: err } })
-			})
-		return () => cancel()
-	}, [embedded, nodeStarted, state.daemonAddress, eventEmitter])
-
-	React.useEffect(() => {
-		if (!state.convsClosed && state.listDone) {
-			dispatch({ type: 'CONVS_CLOSED' })
-			for (const conv of Object.values(state.conversations) as any) {
-				if (conv.isOpen) {
-					state.client.conversationClose({ groupPk: conv.publicKey }).catch((e: any) => {
-						console.warn(`failed to close conversation "${conv.displayName}",`, e)
-					})
-				}
-			}
-		}
-	}, [state.client, state.conversations, state.convsClosed, state.listDone])
+	useEffect(() => initialLaunch(dispatch, embedded), [embedded])
+
+	useEffect(() => openingDaemon(dispatch, state.appState), [embedded, state.appState])
+
+	useEffect(() => openingClients(dispatch, state.appState, eventEmitter, daemonAddress, embedded), [
+		daemonAddress,
+		embedded,
+		eventEmitter,
+		state.appState,
+		state.selectedAccount,
+	])
+
+	useEffect(() => openingListingEvents(state.appState, state.initialListComplete, dispatch), [
+		state.appState,
+		state.initialListComplete,
+	])
+
+	useEffect(() => openingLocalSettings(dispatch, state.appState, state.selectedAccount), [
+		state.appState,
+		state.selectedAccount,
+	])
+
+	useEffect(() => openingCloseConvos(state.appState, dispatch, state.client, state.conversations), [
+		state.appState,
+		state.client,
+		state.conversations,
+	])
+
+	useEffect(() => closingDaemon(state.clearDaemon, state.clearClients, dispatch), [
+		state.clearDaemon,
+		state.clearClients,
+	])
+
+	useEffect(() => deletingStorage(state.appState, dispatch, embedded, state.selectedAccount), [
+		state.appState,
+		state.selectedAccount,
+		embedded,
+	])
 
 	return (
 		<MsgrContext.Provider
 			value={{
 				...state,
-				restart,
-				deleteAccount,
 				dispatch,
-				addNotificationListener,
-				removeNotificationListener,
-				setPersistentOption,
+				addNotificationListener: (cb) => {
+					eventEmitter.addListener('notification', cb)
+				},
+				removeNotificationListener: (cb) => {
+					eventEmitter.removeListener('notification', cb)
+				},
+				restart: () => dispatch({ type: MessengerActions.Restart }),
+				switchToAccount: (accountID: number) =>
+					dispatch({ type: MessengerActions.SetNextAccount, payload: accountID }),
+				deleteAccount: () => dispatch({ type: MessengerActions.SetStateDeleting }),
+
+				setPersistentOption: (key: string, value: any) =>
+					setPersistentOption(dispatch, state, key, value),
+				createNewAccount: () => createNewAccount(embedded, dispatch),
+				importAccount: (path: string) => importAccount(embedded, dispatch, path),
```

Comment:
these needs to be wraped with useCallback to prevent useless rerender and potential inifinite loops